File: DSA/Khushi_590028673/Day27/Q1.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    int compress(vector<char>& chars) {
        int n = chars.size();
        int i = 0;
        int write = 0;

        while (i < n) {
            int j = i;

            while (j < n && chars[j] == chars[i]) {
                j++;
            }

            chars[write++] = chars[i];

            int count = j - i;

            if (count > 1) {
                string s = to_string(count);
                for (char c : s) {
                    chars[write++] = c;
                }
            }

            i = j;
        }

        return write;
    }
};

int main() {
    Solution obj;

    vector<char> chars = {'a','a','b','b','c','c','c'};

    int newLength = obj.compress(chars);

    cout << "Compressed Array: ";
    for (int i = 0; i < newLength; i++) {
        cout << chars[i] << " ";
    }

    cout << "\nLength = " << newLength << endl;

    return 0;
}

File: DSA/vaishnavi_joshi_590016947/Day27/Q1.java
class Solution {
    public int compress(char[] chars) {

        int i = 0;      // Read pointer
        int index = 0;  // Write pointer

        while (i < chars.length) {

            char current = chars[i];
            int count = 0;

            // Count consecutive same characters
            while (i < chars.length && chars[i] == current) {
                count++;
                i++;
            }

            // Write the character
            chars[index++] = current;

            // Write the count if greater than 1
            if (count > 1) {
                String str = String.valueOf(count);

                for (char c : str.toCharArray()) {
                    chars[index++] = c;
                }
            }
        }

        return index;
    }
}